common/log/tag/interface.go

```diff
@@ -0,0 +1,192 @@
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package tag
+
+import (
+	"time"
+)
+
+// Tag is the interface for logging system
+type Tag interface {
+	GetKey() string
+	GetValueType() valueTypeEnum
+
+	GetString() string
+	GetInteger() int64
+	GetDouble() float64
+	GetBool() bool
+	GetError() error
+	GetDuration() time.Duration
+	GetTime() time.Time
+	GetObject() interface{}
+}
+
+var _ Tag = (*tagImpl)(nil)
+
+// the tag value types supported
+const (
+	// ValueTypeString is string value type
+	ValueTypeString valueTypeEnum = 1
+	// ValueTypeInteger is any value of the types(int8,int16,int32,unit8,uint32,uint64) will be converted into int64
+	ValueTypeInteger valueTypeEnum = 2
+	// ValueTypeDouble is for both float and double will be converted into double
+	ValueTypeDouble valueTypeEnum = 3
+	// ValueTypeBool is bool value type
+	ValueTypeBool valueTypeEnum = 4
+	// ValueTypeError is error type value
+	ValueTypeError valueTypeEnum = 5
+	// ValueTypeDuration is duration type value
+	ValueTypeDuration valueTypeEnum = 6
+	// ValueTypeTime is time type value
+	ValueTypeTime valueTypeEnum = 7
+	// ValueTypeObject will be converted into string by fmt.Sprintf("%+v")
+	ValueTypeObject valueTypeEnum = 8
+)
+
+// keep this module private
+type valueTypeEnum int
+
+// keep this module private
+type tagImpl struct {
+	key       string
+	valueType valueTypeEnum
+
+	valueString   string        //1
+	valueInteger  int64         //2
+	valueDouble   float64       //3
+	valueBool     bool          //4
+	valueError    error         //5
+	valueDuration time.Duration //6
+	valueTime     time.Time     //7
+	valueObject   interface{}   //8
+}
+
+func (t *tagImpl) GetKey() string {
+	return t.key
+}
+
+func (t *tagImpl) GetValueType() valueTypeEnum {
+	return t.valueType
+}
+
+func (t *tagImpl) GetString() string {
+	return t.valueString
+}
+
+func (t *tagImpl) GetInteger() int64 {
+	return t.valueInteger
+}
+
+func (t *tagImpl) GetDouble() float64 {
+	return t.valueDouble
+}
+
+func (t *tagImpl) GetBool() bool {
+	return t.valueBool
+}
+
+func (t *tagImpl) GetError() error {
+	return t.valueError
+}
+
+func (t *tagImpl) GetDuration() time.Duration {
+	return t.valueDuration
+}
+
+func (t *tagImpl) GetTime() time.Time {
+	return t.valueTime
+}
+
+func (t *tagImpl) GetObject() interface{} {
```

Comment:
In go naming convention, this is called GetInterface()

Reply:
OK. btw all of those will become GetField()

Reply:
yeah, this comment becomes outdated. On the naming side, one more nit. GetFoo() is not standard go idiom. Go idiom is to just have a method called Field() that returns zap.Field